x-pack/legacy/plugins/alerting/mappings.json

```diff
@@ -4,6 +4,9 @@
       "enabled": {
         "type": "boolean"
       },
+      "tags": {
+        "type": "keyword"
```

Comment:
I've been thinking about "tags" relating to the encrypted SO only-update-no-reindex issue.  If I start with tags `[foo]`, and then update the SO with tags set to `[bar]`, I don't end up with `[foo, bar]` do I?  Perhaps it's worth a test ...

Reply:
Just looked through the code. It doesn't seem an array objects have the same issue. I found a test that makes sure the update only updates with the new values:

- Generates an alert with `['foo']` tag: https://github.com/elastic/kibana/blob/fddac285d04d3f749b1b166db0d9e3ce41810c8f/x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/update.ts#L29
- Sets `['bar']` as the updated tag: https://github.com/elastic/kibana/blob/fddac285d04d3f749b1b166db0d9e3ce41810c8f/x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/update.ts#L34
- Asserts only `['bar']` is there: https://github.com/elastic/kibana/blob/fddac285d04d3f749b1b166db0d9e3ce41810c8f/x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/update.ts#L63

I also just did a manual test doing a GET from the API after just to double check.

Reply:
cool.  The combination of 'partial updates' and ES's notion of single/array values for fields scares me, requiring tags on updates certainly makes this a lot clearer anyway.